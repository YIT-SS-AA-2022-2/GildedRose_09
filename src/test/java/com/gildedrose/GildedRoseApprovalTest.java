package com.gildedrose;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import java.util.Arrays;
import java.util.stream.Collectors;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class GildedRoseApprovalTest {

    @Test
    public void gildedRoseApprovalTest() throws Exception{
        Item[] items = new Item[]{
                new Item("Aged Brie", 20, 0), //
                new Item("Sulfuras", 0, 80), //
                new Item("Sulfuras", -1, 80),
                new Item("Backstage passes", 15, 20),
                new Item("Backstage passes", 10, 20),
                new Item("Backstage passes", 5, 20)
        };
        GildedRose app = new GildedRose(items);
        StringBuilder results = new StringBuilder();
        for(int day = 1; day <= 100; day++){
            String dayAndItems = Arrays.asList(items).stream().map(n->String.valueOf(n)).collect(Collectors.joining("\t"));
            results.append(dayAndItems);
            results.append("\n");
            app.updateQuality();
        }
        Approvals.verify(results.toString());
        assertEquals(41742866, results.toString().hashCode());
    }
}
