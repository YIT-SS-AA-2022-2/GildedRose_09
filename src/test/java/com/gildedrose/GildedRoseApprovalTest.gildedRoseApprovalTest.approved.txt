Aged Brie, 20, 0	Sulfuras, 0, 80	Sulfuras, -1, 80	Backstage passes, 15, 20	Backstage passes, 10, 20	Backstage passes, 5, 20
Aged Brie, 19, 1	Sulfuras, -1, 78	Sulfuras, -2, 78	Backstage passes, 14, 19	Backstage passes, 9, 19	Backstage passes, 4, 19
Aged Brie, 18, 2	Sulfuras, -2, 76	Sulfuras, -3, 76	Backstage passes, 13, 18	Backstage passes, 8, 18	Backstage passes, 3, 18
Aged Brie, 17, 3	Sulfuras, -3, 74	Sulfuras, -4, 74	Backstage passes, 12, 17	Backstage passes, 7, 17	Backstage passes, 2, 17
Aged Brie, 16, 4	Sulfuras, -4, 72	Sulfuras, -5, 72	Backstage passes, 11, 16	Backstage passes, 6, 16	Backstage passes, 1, 16
Aged Brie, 15, 5	Sulfuras, -5, 70	Sulfuras, -6, 70	Backstage passes, 10, 15	Backstage passes, 5, 15	Backstage passes, 0, 15
Aged Brie, 14, 6	Sulfuras, -6, 68	Sulfuras, -7, 68	Backstage passes, 9, 14	Backstage passes, 4, 14	Backstage passes, -1, 13
Aged Brie, 13, 7	Sulfuras, -7, 66	Sulfuras, -8, 66	Backstage passes, 8, 13	Backstage passes, 3, 13	Backstage passes, -2, 11
Aged Brie, 12, 8	Sulfuras, -8, 64	Sulfuras, -9, 64	Backstage passes, 7, 12	Backstage passes, 2, 12	Backstage passes, -3, 9
Aged Brie, 11, 9	Sulfuras, -9, 62	Sulfuras, -10, 62	Backstage passes, 6, 11	Backstage passes, 1, 11	Backstage passes, -4, 7
Aged Brie, 10, 10	Sulfuras, -10, 60	Sulfuras, -11, 60	Backstage passes, 5, 10	Backstage passes, 0, 10	Backstage passes, -5, 5
Aged Brie, 9, 11	Sulfuras, -11, 58	Sulfuras, -12, 58	Backstage passes, 4, 9	Backstage passes, -1, 8	Backstage passes, -6, 3
Aged Brie, 8, 12	Sulfuras, -12, 56	Sulfuras, -13, 56	Backstage passes, 3, 8	Backstage passes, -2, 6	Backstage passes, -7, 1
Aged Brie, 7, 13	Sulfuras, -13, 54	Sulfuras, -14, 54	Backstage passes, 2, 7	Backstage passes, -3, 4	Backstage passes, -8, 0
Aged Brie, 6, 14	Sulfuras, -14, 52	Sulfuras, -15, 52	Backstage passes, 1, 6	Backstage passes, -4, 2	Backstage passes, -9, 0
Aged Brie, 5, 15	Sulfuras, -15, 50	Sulfuras, -16, 50	Backstage passes, 0, 5	Backstage passes, -5, 0	Backstage passes, -10, 0
Aged Brie, 4, 16	Sulfuras, -16, 48	Sulfuras, -17, 48	Backstage passes, -1, 3	Backstage passes, -6, 0	Backstage passes, -11, 0
Aged Brie, 3, 17	Sulfuras, -17, 46	Sulfuras, -18, 46	Backstage passes, -2, 1	Backstage passes, -7, 0	Backstage passes, -12, 0
Aged Brie, 2, 18	Sulfuras, -18, 44	Sulfuras, -19, 44	Backstage passes, -3, 0	Backstage passes, -8, 0	Backstage passes, -13, 0
Aged Brie, 1, 19	Sulfuras, -19, 42	Sulfuras, -20, 42	Backstage passes, -4, 0	Backstage passes, -9, 0	Backstage passes, -14, 0
Aged Brie, 0, 20	Sulfuras, -20, 40	Sulfuras, -21, 40	Backstage passes, -5, 0	Backstage passes, -10, 0	Backstage passes, -15, 0
Aged Brie, -1, 22	Sulfuras, -21, 38	Sulfuras, -22, 38	Backstage passes, -6, 0	Backstage passes, -11, 0	Backstage passes, -16, 0
Aged Brie, -2, 24	Sulfuras, -22, 36	Sulfuras, -23, 36	Backstage passes, -7, 0	Backstage passes, -12, 0	Backstage passes, -17, 0
Aged Brie, -3, 26	Sulfuras, -23, 34	Sulfuras, -24, 34	Backstage passes, -8, 0	Backstage passes, -13, 0	Backstage passes, -18, 0
Aged Brie, -4, 28	Sulfuras, -24, 32	Sulfuras, -25, 32	Backstage passes, -9, 0	Backstage passes, -14, 0	Backstage passes, -19, 0
Aged Brie, -5, 30	Sulfuras, -25, 30	Sulfuras, -26, 30	Backstage passes, -10, 0	Backstage passes, -15, 0	Backstage passes, -20, 0
Aged Brie, -6, 32	Sulfuras, -26, 28	Sulfuras, -27, 28	Backstage passes, -11, 0	Backstage passes, -16, 0	Backstage passes, -21, 0
Aged Brie, -7, 34	Sulfuras, -27, 26	Sulfuras, -28, 26	Backstage passes, -12, 0	Backstage passes, -17, 0	Backstage passes, -22, 0
Aged Brie, -8, 36	Sulfuras, -28, 24	Sulfuras, -29, 24	Backstage passes, -13, 0	Backstage passes, -18, 0	Backstage passes, -23, 0
Aged Brie, -9, 38	Sulfuras, -29, 22	Sulfuras, -30, 22	Backstage passes, -14, 0	Backstage passes, -19, 0	Backstage passes, -24, 0
Aged Brie, -10, 40	Sulfuras, -30, 20	Sulfuras, -31, 20	Backstage passes, -15, 0	Backstage passes, -20, 0	Backstage passes, -25, 0
Aged Brie, -11, 42	Sulfuras, -31, 18	Sulfuras, -32, 18	Backstage passes, -16, 0	Backstage passes, -21, 0	Backstage passes, -26, 0
Aged Brie, -12, 44	Sulfuras, -32, 16	Sulfuras, -33, 16	Backstage passes, -17, 0	Backstage passes, -22, 0	Backstage passes, -27, 0
Aged Brie, -13, 46	Sulfuras, -33, 14	Sulfuras, -34, 14	Backstage passes, -18, 0	Backstage passes, -23, 0	Backstage passes, -28, 0
Aged Brie, -14, 48	Sulfuras, -34, 12	Sulfuras, -35, 12	Backstage passes, -19, 0	Backstage passes, -24, 0	Backstage passes, -29, 0
Aged Brie, -15, 50	Sulfuras, -35, 10	Sulfuras, -36, 10	Backstage passes, -20, 0	Backstage passes, -25, 0	Backstage passes, -30, 0
Aged Brie, -16, 50	Sulfuras, -36, 8	Sulfuras, -37, 8	Backstage passes, -21, 0	Backstage passes, -26, 0	Backstage passes, -31, 0
Aged Brie, -17, 50	Sulfuras, -37, 6	Sulfuras, -38, 6	Backstage passes, -22, 0	Backstage passes, -27, 0	Backstage passes, -32, 0
Aged Brie, -18, 50	Sulfuras, -38, 4	Sulfuras, -39, 4	Backstage passes, -23, 0	Backstage passes, -28, 0	Backstage passes, -33, 0
Aged Brie, -19, 50	Sulfuras, -39, 2	Sulfuras, -40, 2	Backstage passes, -24, 0	Backstage passes, -29, 0	Backstage passes, -34, 0
Aged Brie, -20, 50	Sulfuras, -40, 0	Sulfuras, -41, 0	Backstage passes, -25, 0	Backstage passes, -30, 0	Backstage passes, -35, 0
Aged Brie, -21, 50	Sulfuras, -41, 0	Sulfuras, -42, 0	Backstage passes, -26, 0	Backstage passes, -31, 0	Backstage passes, -36, 0
Aged Brie, -22, 50	Sulfuras, -42, 0	Sulfuras, -43, 0	Backstage passes, -27, 0	Backstage passes, -32, 0	Backstage passes, -37, 0
Aged Brie, -23, 50	Sulfuras, -43, 0	Sulfuras, -44, 0	Backstage passes, -28, 0	Backstage passes, -33, 0	Backstage passes, -38, 0
Aged Brie, -24, 50	Sulfuras, -44, 0	Sulfuras, -45, 0	Backstage passes, -29, 0	Backstage passes, -34, 0	Backstage passes, -39, 0
Aged Brie, -25, 50	Sulfuras, -45, 0	Sulfuras, -46, 0	Backstage passes, -30, 0	Backstage passes, -35, 0	Backstage passes, -40, 0
Aged Brie, -26, 50	Sulfuras, -46, 0	Sulfuras, -47, 0	Backstage passes, -31, 0	Backstage passes, -36, 0	Backstage passes, -41, 0
Aged Brie, -27, 50	Sulfuras, -47, 0	Sulfuras, -48, 0	Backstage passes, -32, 0	Backstage passes, -37, 0	Backstage passes, -42, 0
Aged Brie, -28, 50	Sulfuras, -48, 0	Sulfuras, -49, 0	Backstage passes, -33, 0	Backstage passes, -38, 0	Backstage passes, -43, 0
Aged Brie, -29, 50	Sulfuras, -49, 0	Sulfuras, -50, 0	Backstage passes, -34, 0	Backstage passes, -39, 0	Backstage passes, -44, 0
Aged Brie, -30, 50	Sulfuras, -50, 0	Sulfuras, -51, 0	Backstage passes, -35, 0	Backstage passes, -40, 0	Backstage passes, -45, 0
Aged Brie, -31, 50	Sulfuras, -51, 0	Sulfuras, -52, 0	Backstage passes, -36, 0	Backstage passes, -41, 0	Backstage passes, -46, 0
Aged Brie, -32, 50	Sulfuras, -52, 0	Sulfuras, -53, 0	Backstage passes, -37, 0	Backstage passes, -42, 0	Backstage passes, -47, 0
Aged Brie, -33, 50	Sulfuras, -53, 0	Sulfuras, -54, 0	Backstage passes, -38, 0	Backstage passes, -43, 0	Backstage passes, -48, 0
Aged Brie, -34, 50	Sulfuras, -54, 0	Sulfuras, -55, 0	Backstage passes, -39, 0	Backstage passes, -44, 0	Backstage passes, -49, 0
Aged Brie, -35, 50	Sulfuras, -55, 0	Sulfuras, -56, 0	Backstage passes, -40, 0	Backstage passes, -45, 0	Backstage passes, -50, 0
Aged Brie, -36, 50	Sulfuras, -56, 0	Sulfuras, -57, 0	Backstage passes, -41, 0	Backstage passes, -46, 0	Backstage passes, -51, 0
Aged Brie, -37, 50	Sulfuras, -57, 0	Sulfuras, -58, 0	Backstage passes, -42, 0	Backstage passes, -47, 0	Backstage passes, -52, 0
Aged Brie, -38, 50	Sulfuras, -58, 0	Sulfuras, -59, 0	Backstage passes, -43, 0	Backstage passes, -48, 0	Backstage passes, -53, 0
Aged Brie, -39, 50	Sulfuras, -59, 0	Sulfuras, -60, 0	Backstage passes, -44, 0	Backstage passes, -49, 0	Backstage passes, -54, 0
Aged Brie, -40, 50	Sulfuras, -60, 0	Sulfuras, -61, 0	Backstage passes, -45, 0	Backstage passes, -50, 0	Backstage passes, -55, 0
Aged Brie, -41, 50	Sulfuras, -61, 0	Sulfuras, -62, 0	Backstage passes, -46, 0	Backstage passes, -51, 0	Backstage passes, -56, 0
Aged Brie, -42, 50	Sulfuras, -62, 0	Sulfuras, -63, 0	Backstage passes, -47, 0	Backstage passes, -52, 0	Backstage passes, -57, 0
Aged Brie, -43, 50	Sulfuras, -63, 0	Sulfuras, -64, 0	Backstage passes, -48, 0	Backstage passes, -53, 0	Backstage passes, -58, 0
Aged Brie, -44, 50	Sulfuras, -64, 0	Sulfuras, -65, 0	Backstage passes, -49, 0	Backstage passes, -54, 0	Backstage passes, -59, 0
Aged Brie, -45, 50	Sulfuras, -65, 0	Sulfuras, -66, 0	Backstage passes, -50, 0	Backstage passes, -55, 0	Backstage passes, -60, 0
Aged Brie, -46, 50	Sulfuras, -66, 0	Sulfuras, -67, 0	Backstage passes, -51, 0	Backstage passes, -56, 0	Backstage passes, -61, 0
Aged Brie, -47, 50	Sulfuras, -67, 0	Sulfuras, -68, 0	Backstage passes, -52, 0	Backstage passes, -57, 0	Backstage passes, -62, 0
Aged Brie, -48, 50	Sulfuras, -68, 0	Sulfuras, -69, 0	Backstage passes, -53, 0	Backstage passes, -58, 0	Backstage passes, -63, 0
Aged Brie, -49, 50	Sulfuras, -69, 0	Sulfuras, -70, 0	Backstage passes, -54, 0	Backstage passes, -59, 0	Backstage passes, -64, 0
Aged Brie, -50, 50	Sulfuras, -70, 0	Sulfuras, -71, 0	Backstage passes, -55, 0	Backstage passes, -60, 0	Backstage passes, -65, 0
Aged Brie, -51, 50	Sulfuras, -71, 0	Sulfuras, -72, 0	Backstage passes, -56, 0	Backstage passes, -61, 0	Backstage passes, -66, 0
Aged Brie, -52, 50	Sulfuras, -72, 0	Sulfuras, -73, 0	Backstage passes, -57, 0	Backstage passes, -62, 0	Backstage passes, -67, 0
Aged Brie, -53, 50	Sulfuras, -73, 0	Sulfuras, -74, 0	Backstage passes, -58, 0	Backstage passes, -63, 0	Backstage passes, -68, 0
Aged Brie, -54, 50	Sulfuras, -74, 0	Sulfuras, -75, 0	Backstage passes, -59, 0	Backstage passes, -64, 0	Backstage passes, -69, 0
Aged Brie, -55, 50	Sulfuras, -75, 0	Sulfuras, -76, 0	Backstage passes, -60, 0	Backstage passes, -65, 0	Backstage passes, -70, 0
Aged Brie, -56, 50	Sulfuras, -76, 0	Sulfuras, -77, 0	Backstage passes, -61, 0	Backstage passes, -66, 0	Backstage passes, -71, 0
Aged Brie, -57, 50	Sulfuras, -77, 0	Sulfuras, -78, 0	Backstage passes, -62, 0	Backstage passes, -67, 0	Backstage passes, -72, 0
Aged Brie, -58, 50	Sulfuras, -78, 0	Sulfuras, -79, 0	Backstage passes, -63, 0	Backstage passes, -68, 0	Backstage passes, -73, 0
Aged Brie, -59, 50	Sulfuras, -79, 0	Sulfuras, -80, 0	Backstage passes, -64, 0	Backstage passes, -69, 0	Backstage passes, -74, 0
Aged Brie, -60, 50	Sulfuras, -80, 0	Sulfuras, -81, 0	Backstage passes, -65, 0	Backstage passes, -70, 0	Backstage passes, -75, 0
Aged Brie, -61, 50	Sulfuras, -81, 0	Sulfuras, -82, 0	Backstage passes, -66, 0	Backstage passes, -71, 0	Backstage passes, -76, 0
Aged Brie, -62, 50	Sulfuras, -82, 0	Sulfuras, -83, 0	Backstage passes, -67, 0	Backstage passes, -72, 0	Backstage passes, -77, 0
Aged Brie, -63, 50	Sulfuras, -83, 0	Sulfuras, -84, 0	Backstage passes, -68, 0	Backstage passes, -73, 0	Backstage passes, -78, 0
Aged Brie, -64, 50	Sulfuras, -84, 0	Sulfuras, -85, 0	Backstage passes, -69, 0	Backstage passes, -74, 0	Backstage passes, -79, 0
Aged Brie, -65, 50	Sulfuras, -85, 0	Sulfuras, -86, 0	Backstage passes, -70, 0	Backstage passes, -75, 0	Backstage passes, -80, 0
Aged Brie, -66, 50	Sulfuras, -86, 0	Sulfuras, -87, 0	Backstage passes, -71, 0	Backstage passes, -76, 0	Backstage passes, -81, 0
Aged Brie, -67, 50	Sulfuras, -87, 0	Sulfuras, -88, 0	Backstage passes, -72, 0	Backstage passes, -77, 0	Backstage passes, -82, 0
Aged Brie, -68, 50	Sulfuras, -88, 0	Sulfuras, -89, 0	Backstage passes, -73, 0	Backstage passes, -78, 0	Backstage passes, -83, 0
Aged Brie, -69, 50	Sulfuras, -89, 0	Sulfuras, -90, 0	Backstage passes, -74, 0	Backstage passes, -79, 0	Backstage passes, -84, 0
Aged Brie, -70, 50	Sulfuras, -90, 0	Sulfuras, -91, 0	Backstage passes, -75, 0	Backstage passes, -80, 0	Backstage passes, -85, 0
Aged Brie, -71, 50	Sulfuras, -91, 0	Sulfuras, -92, 0	Backstage passes, -76, 0	Backstage passes, -81, 0	Backstage passes, -86, 0
Aged Brie, -72, 50	Sulfuras, -92, 0	Sulfuras, -93, 0	Backstage passes, -77, 0	Backstage passes, -82, 0	Backstage passes, -87, 0
Aged Brie, -73, 50	Sulfuras, -93, 0	Sulfuras, -94, 0	Backstage passes, -78, 0	Backstage passes, -83, 0	Backstage passes, -88, 0
Aged Brie, -74, 50	Sulfuras, -94, 0	Sulfuras, -95, 0	Backstage passes, -79, 0	Backstage passes, -84, 0	Backstage passes, -89, 0
Aged Brie, -75, 50	Sulfuras, -95, 0	Sulfuras, -96, 0	Backstage passes, -80, 0	Backstage passes, -85, 0	Backstage passes, -90, 0
Aged Brie, -76, 50	Sulfuras, -96, 0	Sulfuras, -97, 0	Backstage passes, -81, 0	Backstage passes, -86, 0	Backstage passes, -91, 0
Aged Brie, -77, 50	Sulfuras, -97, 0	Sulfuras, -98, 0	Backstage passes, -82, 0	Backstage passes, -87, 0	Backstage passes, -92, 0
Aged Brie, -78, 50	Sulfuras, -98, 0	Sulfuras, -99, 0	Backstage passes, -83, 0	Backstage passes, -88, 0	Backstage passes, -93, 0
Aged Brie, -79, 50	Sulfuras, -99, 0	Sulfuras, -100, 0	Backstage passes, -84, 0	Backstage passes, -89, 0	Backstage passes, -94, 0
